pubsub/redis/redis.go

```diff
@@ -352,7 +352,7 @@ func (r *redisStreams) Init(metadata pubsub.Metadata) error {
 		r.client = redis.NewClient(options)
 	}
 
-	if _, err = r.client.Ping().Result(); err != nil {
+	if _, err = r.client.Ping(context.Background()).Result(); err != nil {
```

Comment:
If you move this below the context.WithCancel, you can use `r.ctx` instead of `context.Background()`

Reply:
I've finish that.